awx/main/analytics/core.py

```diff
@@ -122,6 +122,7 @@ def ship(path):
     agent = 'insights-client'
     if shutil.which(agent) is None:
         logger.error('could not find {} on PATH'.format(agent))
+        os.remove(path)
```

Comment:
Honestly, a better way to do this might be to just take the existing function and wrap it in a `try: finally: i.e.,

```
try:
    <current code>
finally:
    os.remove(path)